jerry-core/ecma/builtin-objects/ecma-builtin-object.c

```diff
@@ -669,16 +698,31 @@ ecma_builtin_object_object_get_own_property_descriptor (ecma_value_t this_arg, /
   /* 1. */
   if (!ecma_is_value_object (arg1))
   {
-    ret_value = ecma_raise_type_error (ECMA_ERR_MSG ("Argument is not an object."));
-    return ret_value;
+    return ecma_raise_type_error (ECMA_ERR_MSG ("Argument is not an object."));
   }
 
   ecma_object_t *obj_p = ecma_get_object_from_value (arg1);
 
   /* 2. */
-  ECMA_TRY_CATCH (name_str_value,
-                  ecma_op_to_string (arg2),
-                  ret_value);
+  ecma_value_t name_str_value;
+
+#ifndef CONFIG_DISABLE_ES2015_SYMBOL_BUILTIN
+  if (ecma_is_value_symbol (arg2))
+  {
+    name_str_value = ecma_copy_value (ECMA_CONVERT_SYMBOL_TO_STRING (arg2));
+  }
+  else
+  {
+#endif /* !CONFIG_DISABLE_ES2015_SYMBOL_BUILTIN */
+    name_str_value = ecma_op_to_string (arg2);
+
+    if (ECMA_IS_VALUE_ERROR (name_str_value))
+    {
+      return name_str_value;
+    }
+#ifndef CONFIG_DISABLE_ES2015_SYMBOL_BUILTIN
+  }
+#endif /* !CONFIG_DISABLE_ES2015_SYMBOL_BUILTIN */
 
   ecma_string_t *name_str_p = ecma_get_string_from_value (name_str_value);
```

Comment:
Maybe a funcion `ecma_get_string_or_symbol_from_value` could be introduced to avoid possible confusion.